### ISSUE TITLE
observations contained in train and validation datasets are outside of specified range

### ISSUE THREAD

**Issue by ryanpmccaffrey** [CIA-Oceanix/TrAISformer#31]
According to the paper:

> We used AIS data
from January 01, 2019 to March 10, 2019 and from March 11,
2019 to March 20, 2019 to train the model and tune the hyperparameters, respectively. The test set comprises AIS data from
March 21, 2019 to March 31, 2019.

Although the test set (`ct_dma_test.pkl`) seems to honor this date range, the train and validation sets (`ct_dma_train.pkl` and `ct_dma_valid.pkl`) seem to contain a significant number of observations from dates later in the year, outside the date ranges outlined above.  Why is that?

**Comment by duonginspace** [comment]
Hello, can you provide some examples?

**Comment by ryanpmccaffrey** [comment]
Some snippets:

![image](https://github.com/CIA-Oceanix/TrAISformer/assets/5515646/754988cf-1d7e-4087-82f2-f4339667523c)
![image](https://github.com/CIA-Oceanix/TrAISformer/assets/5515646/381235be-833e-40b5-9f11-8b786980920c)
![image](https://github.com/CIA-Oceanix/TrAISformer/assets/5515646/563ee443-b4b2-4482-9352-6bd10c239214)
![image](https://github.com/CIA-Oceanix/TrAISformer/assets/5515646/b61d445c-6ef0-4ad2-b44f-3ebdd10745aa)
![image](https://github.com/CIA-Oceanix/TrAISformer/assets/5515646/534305f1-9e48-4f3d-914e-701bc99f80ae)

**Comment by duonginspace** [comment]
You're right. 

There was likely an error in the preprocessing step (possibly related to the date format (yyyy-mm-dd vs. yyyy-dd-mm). I will need to investigate the issue when I have the time.
The preprocessing code is here: https://github.com/CIA-Oceanix/GeoTrackNet/blob/master/data/csv2pkl.py

IMPORTANT: I've just checked, there are no test tracks present in the training set.
